### DISCUSSION TITLE
Larger DHCP Pool / Set larger client subnet from /24 to /16

### DISCUSSION THREAD

**Discussion by sysvar** [wg-easy/wg-easy#787] [Q&A] [answered]
**How would you make the server use a /16 subnet rather than /24, (Larger DHCP Pool)?**

I tried this, resulting in crashed app. _(Yes i know its server ip)_
```
    environment:
      - WG_DEFAULT_ADDRESS=192.168.x.x
```

I tried this, resulting in larger 'Allowed IPs' but not server dhcp pool. _(Yes i know its visibility)_
```
    environment:
      - WG_ALLOWED_IPS=192.168.0.0/16
```

Help!

**Comment by eandersons** [comment]
Currently subnet is hardcoded to `/24` in [`src/lib/WireGuard.js`](/WeeJeWel/wg-easy/blob/736c62b4369dc7a6502139a3cbd3a177d3f61727/src/lib/WireGuard.js#L97):

> ```
> Address = ${config.server.address}/24
> ```

One possible workaround is to manually edit `/etc/wireguard/wg0.conf` in the container to replace `Address = 10.0.0.1/24` with `Address = 10.0.0.1/16` in server's configuration, but keep in mind that this change will be overwritten after the container is restarted.

Another and more permanent workaround is to use your own `Dockerfile` to build a custom image that has `Address = ${config.server.address}/16` baked in `lib/WireGuard.js`.
It could look like this (I tested the `sed` command, but not the `Dockerfile` itself):

```Dockerfile
FROM ghcr.io/wg-easy/wg-easy
RUN sed -i 's|\(Address = ${config.server.address}\)/24|\1/16|' /app/lib/WireGuard.js

```

The usage of custom image depends on whether you are running WireGuard Easy container with `docker run` or with Docker Compose, but the Internet has plenty information on this.

**Comment by sysvar** [comment, marked as answer]
Excellent thank you @eandersons, you gave me the needed hints!

I can see that within that file iptables and client side conf template also need the update.

### 1. Explore Values to Update
```
docker exec -it wg-easy cat /app/lib/WireGuard.js | grep "/24"
```

### 2. Implement Final solution
In a Dockerfile, it would look like:
```
FROM ghcr.io/wg-easy/wg-easy
RUN sed -i 's|/24|/16|g' /app/lib/WireGuard.js
```
### 3. Restart & Working
```
docker compose down && docker compose build && docker compose up -d
```

### 4. Final Check
```
docker exec -it wg-easy cat /app/lib/WireGuard.js | grep "/24"
docker exec -it wg-easy cat /app/lib/WireGuard.js | grep "/16"
```

Do make sure AllowedIPs in docker-compose.yaml permits what you expect.
